### PR TITLE
VSCode - Document v2.5 features and edit/restructure content

### LINKED ISSUE #1297: Document Truffle for VSCode walkthrough
Document the Truffle walkthrough on the VSCode getting started splash.

See the dev PR: https://github.com/trufflesuite/vscode-ext/pull/148
See this short video for more details: https://www.loom.com/share/d646f00468774211b624d83214059b18

### LINKED ISSUE #1298: Document VSCode debug network selection
Enhance debugging documentation to include that you can choose which network to debug against.

See the dev PRs: https://github.com/trufflesuite/vscode-ext/pull/170 and https://github.com/trufflesuite/vscode-ext/pull/186
See the second half of this short video for more information: https://www.loom.com/share/d646f00468774211b624d83214059b18

### PR DESCRIPTION
**Pull request by alexandratran**
Edit and restructure VSCode extension content, and document features added in v2.5.

fixes #1297 
fixes #1298

### CODE REVIEW

**Review by kevinbluer — CHANGES_REQUESTED** [review]
Hey @alexandratran, few minor tweaks, although overall it's looking great 🙏

**Comment by OnlyOneJMJQ** [comment]
Does mkdocs take care of 501 redirects or do we need to add those manually?

**Comment by kevinbluer** [comment]
We're using a plugin ([mkdocs-redirects](https://github.com/datarobot/mkdocs-redirects)) to handle them, example of how they're specified [here](https://github.com/trufflesuite/trufflesuite.com/blob/136f23372f7d1e6fcdef2c3ad62b6e6c7c7bd483/mkdocs.yml#L75).

File: src/docs/vscode-ext/how-to/add-a-wallet.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    development: {
+      host: "127.0.0.1",     // Localhost (default: none)
+      port: 8545,            // Standard Ethereum port (default: none)
+      network_id: "*",       // Any network (default: none)
+    }
+    ```
+
+1. To deploy to another target, no other configuration is needed.
+    When selecting deploy contracts, the relevant network section is added dynamically to the Truffle
+    configuration.
```
Comment (rolandtyler) [inline src/docs/vscode-ext/how-to/add-a-wallet.md:51]: I'm getting hung up on "deploy contracts". Are the users selecting something called **deploy contracts** or are they selecting "contracts to deploy"? Or should it be "to deploy contracts?"

File: src/docs/vscode-ext/how-to/add-a-wallet.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+1. To deploy to another target that has been manually added to the Truffle configuration, the
+    provider must be added manually.
```
Comment (rolandtyler) [inline src/docs/vscode-ext/how-to/add-a-wallet.md:54]:
```suggestion
1. To deploy to another target that has been manually added to the Truffle configuration, you must add the
    provider manually.
```
Avoiding passive voice

File: src/docs/vscode-ext/quickstart.md [hunk 3]
```diff
@@ -1,51 +1,71 @@
...
+directly from the VSCode terminal.
 
-## Build your Solidity Contract
+You can also use Truffle for VSCode to scaffold out a project directory using the following steps:
 
-Your newly scaffolded out directory includes a simple contract and all the necessary files to build and deploy a simple, working, contract to the Truffle. Use the following steps to build your contract.
+1. In the [command palette](reference/command-palette.md), select **Truffle: New Solidity Project**.
+1. Select an empty directory to scaffold out your project.
+1. Choose a name for your contract.
+1. Press enter.
```
Comment (rolandtyler) [inline src/docs/vscode-ext/quickstart.md:42]:
```suggestion
1. Press **enter**.
```

File: src/docs/vscode-ext/reference/command-palette.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    .md-typeset table:not([class]) {
+        background-color: var(--md-primary-fg-color);
+    }
+</style>
+
+# Command palette
+
+Truffle for VSCode provides several commands for interacting with ledgers, and building, compiling,
+and testing your smart contracts.
+These features are enabled through the command palette and several "right-click" shortcut menus.
```
Comment (rolandtyler) [inline src/docs/vscode-ext/reference/command-palette.md:11]: Both "right-click" and "right click" are used. WSG uses "right-click" as the action.

**Review by rolandtyler — APPROVED** [review]
A few small comments, but nothing breaking. LGTM

File: src/docs/vscode-ext/how-to/debug-smart-contracts.md [hunk 5]
```diff
@@ -12,17 +12,24 @@ Start the debugger by selecting **Truffle: Debug Transaction** in the
...
-hashes to debug.
+![Debugger](../images/debugger.png)
 
-To debug the other functions (for example, `SendRequest` and `SendResponse` in the
-`HelloBlockchain.sol` contract included in the [basic project](manage-smart-contracts.md)), first
-execute those functions by
+When you initially deploy a contract, only the constructor function is called, so only the
+constructor function is available in the list of transaction hashes to debug.
+To debug other functions (for example, the `sendCoin` function in the `MetaCoin.sol` contract
+included in the [Metacoin box](../../../boxes/metacoin/index.md)), you must invoke them first
```
Comment (kevinbluer) [inline src/docs/vscode-ext/how-to/debug-smart-contracts.md:29]: This link to the box is broken (just needs to be `/boxes/metacoin`)

**Review by kevinbluer — APPROVED** [review]
Other than the link issue it's good to go 👍

**Pull request merged**